### PR TITLE
Refactor product selection dialogs

### LINKED ISSUE #3487: Improve product selection widget
Improve the product selection widget that is used across the Dashboard 2.0 e.g.:
- adding products to a draft order
- adding products to collections or categories

Mockup:
image

### PR DESCRIPTION
**Pull request by dominik-zeglen**
Resolves #3487 

### Pull Request Checklist

1. [ ] Privileged views and APIs are guarded by proper permission checks.
1. [ ] All visible strings are translated with proper context.
1. [ ] All data-formatting is locale-aware (dates, numbers, and so on).
1. [ ] Database queries are optimized and the number of queries is constant.
1. [ ] Database migration files are up to date.
1. [ ] The changes are tested.
1. [ ] The code is documented (docstrings, project documentation).
1. [ ] GraphQL schema and type definitions are up to date.
1. [ ] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by maarcingebala** [comment]
Worth sharing a screenshot of how our new and fancy product selection looks like:
image

**Comment by maarcingebala** [comment]
I've tested it and it looks and works great! One change that I'd suggest is to render a short message if no products were found.

image

**Comment by selvara80** [comment]
As well, good to have this at below as well

Orders/Components/OrderProductAddDialog


regards

File: saleor/static/dashboard-next/orders/components/OrderProductAddDialog/OrderProductAddDialog.tsx [hunk 1]
```diff
@@ -71,25 +139,187 @@ const OrderProductAddDialog = withStyles(styles, {
...
+                          <TableRow>
+                            <TableCell colSpan={4}>
+                              {i18n.t("No products matching given query")}
+                            </TableCell>
+                          </TableRow>
+                        )
+                      )}
+                    </TableBody>
+                  </Table>
+                </InfiniteScroll>
```
Comment (piotrgrundas) [inline saleor/static/dashboard-next/orders/components/OrderProductAddDialog/OrderProductAddDialog.tsx]: You should really think about some code organizing. It's extremally hard to read IMO.
ruciax1

Comment (maarcingebala) [reply saleor/static/dashboard-next/orders/components/OrderProductAddDialog/OrderProductAddDialog.tsx]: True, for instances there are two large `onChange` handlers for checkboxes, shouldn't these be extracted to some functions for better readability?

Comment (piotrgrundas) [reply saleor/static/dashboard-next/orders/components/OrderProductAddDialog/OrderProductAddDialog.tsx]: You could extract functions etc to the top to make `.jsx` a lot clearer.

**Pull request merged**